File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& q) {
        vector<int> ans;
        string vw = "aeiou";
        vector<int> prefix;
        for (string& w : words) {
            if (vw.find(w.front()) != string::npos && vw.find(w.back()) != string::npos) {
                prefix.push_back(1);
            } else {
                prefix.push_back(0);
            }
        }
        
        vector<int> ac(prefix.size());
        partial_sum(prefix.begin(), prefix.end(), ac.begin());
        for (auto& query : q) {
            int x = query[0], y = query[1];
            if (x == 0) {
                ans.push_back(ac[y]);
            } else {
                ans.push_back(ac[y] - ac[x - 1]);
            }
        }
        
        return ans;
    }
};

File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], q: List[List[int]]) -> List[int]:
        ans=[]
        vw="aeiou"
        prefix = [1 if (w[0] in vw and w[-1] in vw) else 0 for w in words]
        ac=list(accumulate(prefix,operator.add))
        
        for x, y in q:
            if x == 0:
                ans.append(ac[y])
            else:
                ans.append(ac[y] - ac[x - 1])
        
        return ans